build/js/quicklink.js

```diff
@@ -1,49 +1,82 @@
-import quicklink from 'quicklink/dist/quicklink.mjs';
+import * as quicklink from 'quicklink';
+
+// Move quicklink to the global scope
+window.quicklink = quicklink;
 
 window.addEventListener( 'load', () => {
 	const exportedOptions = window.quicklinkOptions || {};
+	const prefetchQueue = window.quicklinkPrefetch || {};
 
-	const options = {};
+	const listenerOptions = {};
 
 	// el: Convert selector into element reference.
 	if ( 'string' === typeof exportedOptions.el && exportedOptions.el ) {
-		options.el = document.querySelector( exportedOptions.el );
-	}
-
-	// urls: Verify we don't get an empty array, as that would turn off quicklink.
-	if ( Array.isArray( exportedOptions.urls ) && 0 < exportedOptions.urls.length ) {
-		options.urls = exportedOptions.urls;
+		listenerOptions.el = document.querySelector( exportedOptions.el );
 	}
 
 	// timeout: Verify we actually get an int for milliseconds.
 	if ( 'number' === typeof exportedOptions.timeout ) {
-		options.timeout = exportedOptions.timeout;
+		listenerOptions.timeout = exportedOptions.timeout;
+	}
+
+	// limit: Verify we actually get an int.
+	if ( 'number' === typeof exportedOptions.limit && exportedOptions.limit > 0 ) {
+		listenerOptions.limit = exportedOptions.limit;
+	}
+
+	// throttle: Verify we actually get an int.
+	if ( 'number' === typeof exportedOptions.throttle && exportedOptions.throttle > 0 ) {
+		listenerOptions.throttle = exportedOptions.throttle;
 	}
 
 	// timeoutFn: Obtain function reference as opposed to function string, if it is not the default.
-	if ( 'string' === typeof exportedOptions.timeoutFn && 'requestIdleCallback' !== exportedOptions.timeoutFn ) {
+	if ( 'string' === typeof exportedOptions.timeoutFn && 'requestIdleCallback' !== exportedOptions.timeoutFn && typeof 'function' === window[ exportedOptions.timeoutFn ] ) {
 		const timeoutFn = window[ exportedOptions.timeoutFn ];
-		options.timeoutFn = function() {
+		listenerOptions.timeoutFn = function() {
 			return timeoutFn.apply( window, arguments );
 		};
 	}
 
+	// onError: Obtain function reference as opposed to function string, if it is not the default.
+	if ( 'string' === typeof exportedOptions.onError && 'requestIdleCallback' !== exportedOptions.onError && typeof 'function' === window[ exportedOptions.onError ] ) {
```

Comment:
Since the default for this is not `requestIdleCallback` like it is for `timeoutFn`:

```suggestion
	if ( 'string' === typeof exportedOptions.onError && typeof 'function' === window[ exportedOptions.onError ] ) {
```